pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -338,14 +338,14 @@ def run_evaluation(self, test_mode: bool = False):
 
         # select dataloaders
         if test_mode:
-            if self.reload_dataloaders_every_epoch or self.test_dataloaders is None:
+            if self.reload_train_dataloader_every_epoch or self.test_dataloaders is None:
```

Comment:
this is wrong no? don’t we want to reset the test and val dataloaders with every call to evaluate?

Reply:
I don't think we want to ever reset the val dataloader as that would defeat the purpose of tracking progress with a validation set; but in any case I believe I haven't changed any of the functionality as the validation set wasn't being reset before.

Reply:
If we want to keep that, maybe the compromise is to keep the name `reload_dataloaders_every_epoch`, and consider that this reloads the train dataloader every training epoch in `training_loop`, and the val/test dataloaders at every evaluation in `evaluation_loop`. I can just revert the last changes in this case, and the previously-approved PR should be good to go. Sorry if I'm reinventing the wheel here !

Reply:
I thought the point was just that the reload_train_dataloader_every_epoch was doing stuff with test and val when it wasn't needed - not to revert the change?

Reply:
I'm actually not sure anymore we're on the same page here ! But in any case I think it's better to have `reload_dataloaders_every_epoch` stand for everything, as it keeps previous functionality and doesn't split it into an argument for everything (ie train val and test)

Reply:
evaluate runs the test and val loop.

Maybe i'm missing something, but if the user wants to reload the validation and test datasets every time evaluation is checked that should be allowed no?

Reply:
Haha, I'm confused. Originally tbe point was made that `reload_dataloaders_every_epoch` didn't previously reload the test and val dataloaders, only train. So then I suggested the name change, but it turns out we do reload them? Anyway, let's make it so that `reload_dataloaders_every_epoch` does what it says on the tin and applies to val and test aswell, in which case I think the above comment from @tullie still applies. Can then revisit it if someone finds a use case where that doesn't work for them :)

Reply:
@ethanwharris is this also solved?

Reply:
It should be fixed now, but the CI says some checks were cancelled and I'm not sure why :/